src/transactions/ClawbackOpFrame.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/ClawbackOpFrame.h"
+#include "ledger/LedgerTxn.h"
+#include "transactions/TransactionUtils.h"
+#include <Tracy.hpp>
+
+namespace stellar
+{
+
+ClawbackOpFrame::ClawbackOpFrame(Operation const& op, OperationResult& res,
+                                 TransactionFrame& parentTx)
+    : OperationFrame(op, res, parentTx)
+    , mClawback(mOperation.body.clawbackOp())
+    , mAsset(getAsset())
+{
+}
+
+bool
+ClawbackOpFrame::isVersionSupported(uint32_t protocolVersion) const
+{
+    return protocolVersion >= 16;
+}
+
+bool
+ClawbackOpFrame::doApply(AbstractLedgerTxn& ltx)
+{
+    ZoneNamedN(applyZone, "ClawbackOp apply", true);
+
+    LedgerKey key(TRUSTLINE);
+    key.trustLine().accountID = toAccountID(mClawback.from);
+    key.trustLine().asset = mAsset;
+
+    auto trust = ltx.load(key);
```

Comment:
Use `loadTrustLine` instead

Reply:
If we use `loadTrustLine`, `addBalanceSkipAuthorization` and `isClawbackEnabled` will need to be added to `TrustLineWrapper`. This isn't a problem, but I wonder if this is worth doing since these methods are unique to clawback. They also wouldn't make sense in the context of `TrustLineWrapper::IssuerImpl`.

Reply:
Oh hm, that's a good point. Then perhaps `ltx.load(trustlineKey(...))`?

Reply:
Done

Reply:
While working on `ClawbackClaimableBalanceOp`, I had to add `isClawbackEnabled` to `TrustLineWrapper` because it's  used in `CreateClaimableBalanceOp`. I think we should still use `ltx.load(trustlineKey(...))` here though to avoid adding `addBalanceSkipAuthorization` to the wrapper.